In [1]:
#Celda para librerías


import sklearn as sk

import matplotlib as plt
import matplotlib.pyplot as plt

import pandas as pd
import seaborn as sns
import numpy as np
import plotly.graph_objects as go

import warnings
warnings.filterwarnings('ignore')

print(np.__version__)
from ydata_profiling import ProfileReport
import os
print("Directorio actual:", os.getcwd())



# Have all columns appear when dataframes are displayed.
pd.set_option('display.max_columns', None) 
# Have 100 rows appear when a dataframe is displayed
pd.set_option('display.max_rows', 500)
# Display dimensions whenever a dataframe is printed out.
pd.set_option('display.show_dimensions', True)

bureau = pd.read_csv(r'C:/Users/Yeray/Desktop/DATA_SCIENCE_ML/Home-Credit-TFG/DATA/home-credit-default-risk/bureau.csv')
bureau_balance = pd.read_csv(r'C:/Users/Yeray/Desktop/DATA_SCIENCE_ML/Home-Credit-TFG/DATA/home-credit-default-risk/bureau_balance.csv')

1.23.5
Directorio actual: c:\Users\Yeray\Desktop\DATA_SCIENCE_ML\Home-Credit-TFG\JUPYTER_NOTEBOOKS


In [2]:
print(bureau_balance.head(10))
print(bureau_balance.shape)
print(bureau_balance.info())
print(bureau_balance.describe())

   SK_ID_BUREAU  MONTHS_BALANCE STATUS
0       5715448               0      C
1       5715448              -1      C
2       5715448              -2      C
3       5715448              -3      C
4       5715448              -4      C
5       5715448              -5      C
6       5715448              -6      C
7       5715448              -7      C
8       5715448              -8      C
9       5715448              -9      0

[10 rows x 3 columns]
(27299925, 3)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27299925 entries, 0 to 27299924
Data columns (total 3 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   SK_ID_BUREAU    int64 
 1   MONTHS_BALANCE  int64 
 2   STATUS          object
dtypes: int64(2), object(1)
memory usage: 624.8+ MB
None
       SK_ID_BUREAU  MONTHS_BALANCE
count  2.729992e+07    2.729992e+07
mean   6.036297e+06   -3.074169e+01
std    4.923489e+05    2.386451e+01
min    5.001709e+06   -9.600000e+01
25%    5.730933e+06   -4.600000e+01
50%    

In [4]:
#Vamos a generar un reporte de perfilamiento de los datos
profile = ProfileReport(bureau_balance, title="Pandas Profiling Report")
profile.to_file("bureau_balance.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [3]:

bureau_SK_ID_CURR_AND_SK_ID_BUREAU = bureau[['SK_ID_CURR', 'SK_ID_BUREAU']]
print("Valores únicos de SK_ID_BUREAU en bureau:", bureau['SK_ID_BUREAU'].nunique())
print("Valores únicos de SK_ID_BUREAU en bureau_balance:", bureau_balance['SK_ID_BUREAU'].nunique())
unique_bureau_ids = pd.Series(list(set(bureau_SK_ID_CURR_AND_SK_ID_BUREAU['SK_ID_BUREAU'])
                                   .intersection(set(bureau_balance['SK_ID_BUREAU']))))
print("Valores únicos de SK_ID_BUREAU en ambas tablas: ", unique_bureau_ids.shape[0])


Valores únicos de SK_ID_BUREAU en bureau: 1716428
Valores únicos de SK_ID_BUREAU en bureau_balance: 817395
Valores únicos de SK_ID_BUREAU en ambas tablas:  774354


Tenemos 817395 SK_ID_BUREAU valores únicos en bureau_balance mientras que en bureau tenemos 1716428 valores únicos.
Por otro lado, hay 43041 valores SK_ID_BUREAU que se encuentran en la tabla bureau_balance que no están en bureau. Esto significa que no hay IDs que se puedan asociar a esos valores. Estos valores posiblemente serán eliminados pues no se pueden asociar a ningún cliente.

In [4]:
print(bureau_balance['STATUS'].value_counts())

STATUS
C    13646993
0     7499507
X     5810482
1      242347
5       62406
2       23419
3        8924
4        5847
Name: count, Length: 8, dtype: int64
